.github/workflows/docs-diff.yml

```diff
@@ -0,0 +1,190 @@
+      ### Build Python package
+      - name: Build Python
+        run: make python-build
+      ### Build docs
+      - name: Setup Hugo
+        uses: peaceiris/actions-hugo@v2
+        with:
+          hugo-version: '0.79.1'
+      - name: Build Hugo docs
+        run: make build
+        working-directory: docs
+      - name: Upload old build of docs.
+        uses: actions/upload-artifact@v2
+        with:
+          name: old-docs
+          path: docs/public
+  diff:
+    name: Diff
+    needs: [build-head, build-base]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Pull old build of docs.
+        uses: actions/download-artifact@v2
+        with:
+          name: old-docs
+          path: base
+      - name: Pull new build of docs.
+        uses: actions/download-artifact@v2
+        with:
+          name: new-docs
+          path: head
+      - name: Remove ignored files (generated api docs)
+        run: rm -rf base/java/reference/api base/python/reference/api base/ruby/reference/api base/node/reference/api
+      - name: Remove ignored files (generated api docs)
+        run: rm -rf head/java/reference/api head/python/reference/api head/ruby/reference/api head/node/reference/api
+
+      - name: Diff
+        id: diff
+        run: set -o pipefail && diff -q --recursive base head | tee diff.txt
+        continue-on-error: true
+      - id: get-comment-body
+        run: |
+          body=$(cat <<EOF
+            This PR contains docs changes. Comment "Docs changes ok" and rerun the job
+            if you expected this, and review the changes by running the docs locally,
+            checking for out of date code snippets.
+
+            Diff:
+
+          EOF
+          )
+          body=$(cat <(echo $body) diff.txt)
+          body="${body//'%'/'%25'}"
+          body="${body//$'\n'/'%0A'}"
+          body="${body//$'\r'/'%0D'}"
+          echo "::set-output name=body::$body"
+        if: ${{ steps.diff.outcome == 'failure' }}
+      - name: Look for bot comment.
+        id: comment-bot-check
+        uses: peter-evans/find-comment@v1
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body-includes: "Docs changes ok"
+          comment-author: "github-actions[bot]"
+        if: ${{ steps.diff.outcome == 'failure' }}
+      - name: Comment
+        uses: peter-evans/create-or-update-comment@v1
+        with:
+          comment-id: ${{ steps.comment-bot-check.outputs.comment-id }}
+          issue-number: ${{ github.event.pull_request.number }}
+          body: ${{ steps.get-comment-body.outputs.body }}
+          edit-mode: replace
+        if: ${{ steps.diff.outcome == 'failure' }}
+      - name: Look for resolving comment.
+        id: comment-check
+        uses: peter-evans/find-comment@v1
+        with:
+          issue-number: ${{ github.event.pull_request.number }}
+          body-includes: "Docs changes ok"
+          comment-author: ${{ github.event.pull_request.user.login }}
```

Comment:
This might be annoying if someone wants to help someone else get something merged?

Reply:
In those cases I think we could force the merge with administrator privileges. They don't happen too frequently. If the PR author intended to make docs changes they can just leave this comment right away.